packages/node/src/integrations/http.ts

```diff
@@ -0,0 +1,161 @@
+import { addBreadcrumb, getCurrentClient } from '@sentry/shim';
```

Comment:
The complexity and code structure of this file is not optimal, can we refactor this somehow, it's really hard to follow what's really happening here.